pennylane/transforms/split_to_single_terms.py

```diff
@@ -0,0 +1,177 @@
+# Copyright 2018-2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Contains the tape transform that splits multi-term measurements on a tape into single-term measurements,
+all included on the same tape. This transform expands sums but does not divide non-commuting measurements
+between different tapes.
+"""
+
+from functools import partial
+
+from pennylane.transforms import transform
+from pennylane.transforms.split_non_commuting import (
+    _processing_fn_no_grouping,
+    _split_all_multi_term_obs_mps,
+)
+
+
+def null_postprocessing(results):
+    """A postprocessing function returned by a transform that only converts the batch of results
+    into a result for a single ``QuantumTape``.
+    """
+    return results[0]
+
+
+@transform
+def split_to_single_terms(tape):
+    """Splits any expectation values of multi-term observables in a circuit into single terms.
+    For devices that don't natively support measuring expectation values of sums of observables.
```

Comment:
Not a blocker, and it's clear to me after reading the docstring, but new users might be confused about when to use `split_non_commuting` and `split_to_single_terms`. Could be worth having a sentence somewhere to explain the distinction.